File: benchmarks/core-comparison/programs/scalar-control.c
#include <stdio.h>

static double advance(double x, double i) {
    double y = x * 1.00000011920929 + i * 0.0000001;
    return y > 1000.0 ? y - 999.5 : y;
}

static double run(double n) {
    double i = 0.0;
    double x = 1.0;
    while (i < n) {
        x = advance(x, i);
        i += 1.0;
    }
    return x;
}

int main(void) {
    volatile double count = {{COUNT}}.0;
    printf("%.17g\n", run(count));
    return 0;
}

File: benchmarks/core-comparison/programs/scalar-control.go
package main

import (
	"fmt"
	"os"
)

func advance(x, i float64) float64 {
	y := x*1.00000011920929 + i*0.0000001
	if y > 1000.0 {
		return y - 999.5
	}
	return y
}

func run(n float64) float64 {
	i := 0.0
	x := 1.0
	for i < n {
		x = advance(x, i)
		i += 1.0
	}
	return x
}

func main() {
	count := float64({{COUNT}} + len(os.Args) - 1)
	fmt.Printf("%.17g\n", run(count))
}
